src/main/java/database/service/JsonDatabaseService.java

```diff
@@ -179,7 +182,28 @@ public <T extends BaseEntity> T getById(Class<? extends BaseEntity> entityClass,
     public <T extends BaseEntity> Iterable<T> getAllRecordsFromTable(Class<? extends BaseEntity> entityClass) {
         Path databasePath = Path.of(getDatabasePath(entityClass));
 
-        return deserializeEntities(entityClass, readDatabaseFile(databasePath));
+        List<T> entities = deserializeEntities(entityClass, readDatabaseFile(databasePath));
+
+        return entities.stream()
+                .skip(settings.getOffset())
+                .limit(settings.getLimit())
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public <T extends BaseEntity> Iterable<T> getAllRecordsFromTable(Class<? extends BaseEntity> entityClass,
+                                                                     int limit, int offset) {
+        if (limit < 0 || limit > 100 || offset < 0) {
```

Comment:
Please use max `limit` value from `settings` and add checking max `offset` value.